### PR TITLE
Add support for macro aliases

### PR DESCRIPTION
**Pull request by pmatilai**
Implement support for multi-level macro aliases, useful for the kind of thing aliases are: shortcuts, added level of indirection here and there for eg choosing between alternatives without copying and so on.

The new %alias macro primitive can be used to define an alias and get the alias definition. The dual functionality is necessary because all other references to the alias get routed to the aliased macro. With the exception of undefine which undefines the alias itself.

### CODE REVIEW

**Comment by pmatilai** [comment]
My immediate use-case for this will be in #2620 but I remember missing such functionality for at least one other purpose. And who knows what fun the packaging community comes up with...

**Comment by pmatilai** [comment]
Added more elaborate documentation. In order to document the aliases I needed to document all sorts of other macro behavior first, which is not a bad thing...

**Comment by pmatilai** [comment]
Rebase on top of the other recent stuff.

**Comment by ffesti** [comment]
I am still a bit puzzled on what this can do that you can't do with a simple macro definition. I mean I have a  rough idea. But may be the docs should make a bit more of an effort to explain what to use in which case.

**Comment by pmatilai** [comment]
It should be more obvious once I rebase the autobuild thing on top of this all.

But shortly, in the autobuild context, we have these __foo_build, __bar_build, __zap_build etc macros defined for various different buildsystems. Those are all usable in standalone format by calling the explicit lengthy name, but when one uses Autobuild: tag, there should be a shortcut to the active build type, eg in the spec you only then need to use %autobuild_install if you manually need to invoke it for whatever reason. That should also simplify the rpm implementation side of the autobuild a bit. And you can't really achieve this (parametric macro semantics and all) with regular macro definitions. One could make a copy, but we don't have support for that either. 

In other words, maybe it helps to think of aliases as a means to achieve "alternatives" in the macro space.

**Comment by pmatilai** [comment]
This will need at least a C API for adding aliases. And maybe look at the bash-style alias thing too.

**Comment by pmatilai** [comment]
Withdrawing for now.
